test/lib/modules/userStore.test.js

```diff
@@ -1953,4 +1953,636 @@ describe('userStore', () => {
+      });
+    });
+
+    it('should prevent recursive requests to the same object that use user context', (done) => {
+      this.storeUserRequest({ useUserContext: true }).revokeRole(1234, 5678, (err, result) => {
+        should.not.exist(result);
+        should.exist(err);
+        err.message.should.eql('UserStoreError');
+        err.description.should.eql('Not Allowed');
+        err.debug.should.eql('Recursive requests to the user store from the user store cannot use user credentials or use Bl');
+        return done();
+      });
+    });
+
+    it('should prevent recursive requests to the same object that use user context and bl', (done) => {
+      this.storeUserRequest({ useUserContext: true, useBl: true }).revokeRole(1234, 5678, (err, result) => {
+        should.not.exist(result);
+        should.exist(err);
+        err.message.should.eql('UserStoreError');
+        err.description.should.eql('Not Allowed');
+        err.debug.should.eql('Recursive requests to the user store from the user store cannot use user credentials or use Bl');
+        return done();
+      });
+    });
+
+    it('should allow recursive requests to the same object that use mastersecret and skip bl', (done) => {
+      nock('https://baas.kinvey.com')
+        .matchHeader('content-type', 'application/json')
+        .matchHeader('x-kinvey-api-version', '3')
+        .matchHeader('x-kinvey-skip-business-logic', 'true')
+        .delete(`/user/${environmentId}/1234/roles/5678`)
+        .basicAuth({
+          user: environmentId,
+          pass: mastersecret
+        })
+        .reply(200);
+
+      this.storeUserRequest().revokeRole('1234', '5678', (err, result) => {
+        should.not.exist(err);
+        should.not.exist(result);
+        return done();
+      });
+    });
+
+    it('should return an error if missing userId', (done) => {
+      this.store().revokeRole((err, result) => {
+        should.not.exist(result);
+        err.message.should.eql('UserStoreError');
+        err.description.should.eql('Bad Request');
+        err.debug.should.eql('userId is required');
+        return done();
+      });
+    });
+
+    it('should return an error if null userId', (done) => {
+      this.store().revokeRole(null, (err, result) => {
+        should.not.exist(result);
+        err.message.should.eql('UserStoreError');
+        err.description.should.eql('Bad Request');
+        err.debug.should.eql('userId is required');
+        return done();
+      });
+    });
+
+    it('should return an error if blank userId', (done) => {
+      this.store().revokeRole('', (err, result) => {
+        should.not.exist(result);
+        err.message.should.eql('UserStoreError');
+        err.description.should.eql('Bad Request');
+        err.debug.should.eql('userId is required');
+        return done();
+      });
+    });
+
+    it('should return an error if missing roleId', (done) => {
+      this.store().revokeRole(1234, (err, result) => {
```

Comment:
If the second `roleId` is required, why is it optional? I.e. why adjust the callback?

Reply:
It's not optional.  The callback is adjusted in order to make it clear that I'm calling back when returning the error.  I'd rather do that than just go `roleId(error);`